gallery_dl/extractor/mastodon.py

```diff
@@ -171,20 +171,16 @@ class MastodonAPI():
     def __init__(self, extractor):
         self.root = extractor.root
         self.extractor = extractor
+        self.headers = {}
 
         access_token = extractor.config("access-token")
         if access_token is None or access_token == "cache":
             access_token = _access_token_cache(extractor.instance)
-        if not access_token:
-            try:
-                access_token = INSTANCES[extractor.category]["access-token"]
-            except (KeyError, TypeError):
-                raise exception.StopExtraction(
-                    "Missing access token.\n"
-                    "Run 'gallery-dl oauth:mastodon:%s' to obtain one.",
-                    extractor.instance)
+        if not access_token and extractor.category in INSTANCES:
+            access_token = INSTANCES[extractor.category]["access-token"]
 
-        self.headers = {"Authorization": "Bearer " + access_token}
+        if access_token:
+            self.headers = {"Authorization": "Bearer " + access_token}
```

Comment:
Why did you remove the *entire* try except block? It is there to account for user-added entries to `INSTANCES` that do not have an `access-token` value (possible with the right config settings *)

I would suggest doing it this way, while also removing `self.headers = {}` from line 174:
(GitHub doesn't support "applying suggestions on deleted lines", it seems)
``` python
        if not access_token:
            try:
                access_token = INSTANCES[extractor.category]["access-token"]
            except (KeyError, TypeError):
                pass
        if access_token:
            self.headers = {"Authorization": "Bearer " + access_token}
        else:
            self.headers = None
```

-------------

(*)

``` json
{
    "extractor": {
        "mastodon": {
            "improved-madness": {
                "root": "https://mastodon.improved-madness.de"
            }
        }
    }
}

```

Reply:
Oh, I see, I didn't realize that was a possibility. I've updated the PR to incorporate your suggestions.

Reply:
Thanks